pkg/binary/envoy/debug/pii.go

```diff
@@ -0,0 +1,72 @@
+package debug
+
+import (
+	"fmt"
+	"bitbucket.org/creachadair/shell"
+	"crypto/sha256"
+)
+
+// filter the array of logs using formatStr and validFieldNames, hash the resulting array of logs and return the final array
+func process(logsArr []string, formatStr string, validFieldNames map[string]bool) (processedLogs []string) {
+	defer func() {
+		if r := recover(); r != nil {
+			fmt.Println("process func panicking, recovering")
+		}
+	} ()
+
+	fieldNames, ok := shell.Split(formatStr)
+
+	if !ok {
+		panic(fmt.Sprint("error in splitting format string: %s", formatStr))
+	}
+
+	testForValidLog :=  func(log string) bool {
+		fields, ok := shell.Split(log)
+		if !ok {
+			panic(fmt.Sprint("error in splitting log: %s", log))
```

Comment:
Like above, we should replace this with a `return []string{}, fmt.Errorf(...)`.

However, in this case we don't want to abort the entire function just because we can't process one log line. Instead, we should just log and `continue`:

```
if !ok {
	log.debugf("skipping log line because we could not split it: %q", log)
	continue
}
```